File: codinGame/PHP/power_of_thor.php
<?php
fscanf(STDIN, "%d %d %d %d", $lightX, $lightY, $thorX, $thorY);

while (TRUE) {
    fscanf(STDIN, "%d", $remainingTurns); // reads remaining turns from standard input

    $dx = $lightX - $thorX <=> 0; // spaceship operator returns -1, 0, or 1
    $dy = $lightY - $thorY <=> 0;

    $thorX += $dx;
    $thorY += $dy;

    echo (($dy > 0 ? 'S' : ($dy < 0 ? 'N' : '')) .
          ($dx > 0 ? 'E' : ($dx < 0 ? 'W' : ''))) . "\n";
}
?>

File: codinGame/C#/power_of_thor.cs
using System;

class Player {
    static void Main(string[] args) {
        string[] inputs = Console.ReadLine().Split(' ');
        int lightX = int.Parse(inputs[0]);
        int lightY = int.Parse(inputs[1]);
        int thorX = int.Parse(inputs[2]);
        int thorY = int.Parse(inputs[3]);

        while (true) {
            int remainingTurns = int.Parse(Console.ReadLine());  // reading remaining turns (not used)

            int dx = Math.Sign(lightX - thorX);
            int dy = Math.Sign(lightY - thorY);

            thorX += dx;
            thorY += dy;

            Console.WriteLine(
                (dy > 0 ? "S" : (dy < 0 ? "N" : "")) +
                (dx > 0 ? "E" : (dx < 0 ? "W" : ""))
            );
        }
    }
}
